src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,79 @@
-export const Pagination = () => {};
+import cn from 'classnames';
+import React from 'react';
+
+type Props = {
+  total: number;
+  itemsPerPage: number;
+  currentPage?: number;
+  onPageChange: (page: number) => void;
+};
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  itemsPerPage,
+  currentPage = 1,
+  onPageChange,
+}) => {
+  const pageCount = Math.max(1, Math.ceil(total / itemsPerPage));
+  const canNext = currentPage < pageCount;
+  const canPrev = currentPage > 1;
+  const pages = Array.from({ length: pageCount }, (_, index) => index + 1);
+
+  const onSelectPage = (page: number) => {
```

Comment:
prefix `on` before the handler name is a bit wrong here.
the prefix `on` is used for props, the prefix `handle` is used for handlers you create inside of component.

So if you have the component which accepts the select page callback, it should look like this:


```suggestion
  const handleSelectPage = (page: number) => {...}
  
  return (
    <Component onSelectPage={handleSelectPage} />
  )
```

more about naming in react you can find [here](https://dev.to/kristiyanvelkov/react-js-naming-convention-lcg)